#Task-1 : on /update_class_group page got following error. Fix it
127.0.0.1 - - [01/Aug/2025 00:57:19] "GET /update_class_group HTTP/1.1" 500 -
Traceback (most recent call last):
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\app.py", line 1498, in __call__
    return self.wsgi_app(environ, start_response)
           ~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^^^^^^^^^
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\app.py", line 1476, in wsgi_app
    response = self.handle_exception(e)
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\app.py", line 1473, in wsgi_app
    response = self.full_dispatch_request()
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\app.py", line 882, in full_dispatch_request
    rv = self.handle_user_exception(e)
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\app.py", line 880, in full_dispatch_request
    rv = self.dispatch_request()
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\app.py", line 865, in dispatch_request
    return self.ensure_sync(self.view_functions[rule.endpoint])(**view_args)  # type: ignore[no-any-return]
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask_login\utils.py", line 290, in decorated_view
    return current_app.ensure_sync(func)(*args, **kwargs)
           ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^^^^^^^^^^^^^^^^
  File "C:\Users\nasim\Documents\git\sims\app\routes.py", line 1238, in update_class_group
    return render_template('update_class_group.html', title='Update Class Group', indexes=indexes)
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\templating.py", line 150, in render_template
    return _render(app, template, context)
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\flask\templating.py", line 131, in _render
    rv = template.render(context)
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\jinja2\environment.py", line 1304, in render
    self.environment.handle_exception()
    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~^^
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\jinja2\environment.py", line 939, in handle_exception
    raise rewrite_traceback_stack(source=source)
  File "C:\Users\nasim\Documents\git\sims\app\templates\update_class_group.html", line 1, in top-level template code
    {% extends "base.html" %}
  File "C:\Users\nasim\Documents\git\sims\app\templates\base.html", line 237, in top-level template code
    {% block content %}{% endblock %}
    ^^^^^^^^^^^^^^^^^
  File "C:\Users\nasim\Documents\git\sims\app\templates\update_class_group.html", line 23, in block 'content'
    <td>{{ group.name }}</td>
    ^^^^^^^^^
  File "c:\Users\nasim\Documents\git\sims\.venv\Lib\site-packages\jinja2\environment.py", line 487, in getattr
    return getattr(obj, attribute)
jinja2.exceptions.UndefinedError: 'group' is undefined


#Task 2: Under Search Nav menu add a submenuItem 'Class Group' to search data with  list all the fields of ClassGroup to where partial string matches with name or description field
